source/common/http/filter/lua/lua_filter.cc

```diff
@@ -261,8 +261,22 @@ int StreamHandleWrapper::luaHeaders(lua_State* state) {
   if (headers_wrapper_.get() != nullptr) {
     headers_wrapper_.pushStack();
   } else {
-    headers_wrapper_.reset(
-        HeaderMapWrapper::create(state, headers_, [this]() { return !headers_continued_; }), true);
+    headers_wrapper_.reset(HeaderMapWrapper::create(state, headers_,
+                                                    [this] {
+                                                      // If we are about to do a modifiable header
+                                                      // operation, blow away the route cache. We
+                                                      // could be a little more intelligent about
+                                                      // when we do this so the performance would be
+                                                      // higher, but this is simple and will get the
+                                                      // job done for now. This is a NOP on the
+                                                      // encoder path.
+                                                      if (!headers_continued_) {
```

Comment:
Should this also be done in regular filters? I.e. is there a reason to make Lua behave non-uniform here, such as keeping is simple?

Reply:
The reason is keeping it simple. It would be a more complex and performance sensitive change to do it in all filters. I can open an issue to track this idea if you like.

Reply:
Do we have a doc for filter authors that provides advice on how to do things like a rerouting filter? I think just capturing the idioms for Lua vs. C++ would be useful somewhere, issues is fine.